examples/client.rs

```diff
@@ -49,12 +54,30 @@ async fn main() {
                 let addr = format!("{host}:443");
                 let tcp_stream = TcpStream::connect(addr).await.unwrap();
                 let tls_stream = connector.connect(dnsname, tcp_stream).await.unwrap();
-                break MaybeTlsStream::Tls(TlsStream::Rustls(tls_stream));
+                break MaybeTlsStream::from(tls_stream);
+            }
+            #[cfg(feature = "async-std-native-tls")]
+            "tls" => {
+                let addr = format!("{host}:443");
+                let tcp_stream = TcpStream::connect(&addr).await.unwrap();
+                let tls_stream = async_native_tls::connect(host, tcp_stream).await.unwrap();
+                break MaybeTlsStream::from(tls_stream);
+            }
+            #[cfg(feature = "async-std-rustls")]
+            "tls" => {
+                let addr = format!("{host}:443");
+                let tcp_stream = TcpStream::connect(&addr).await.unwrap();
+                let connector = async_tls::TlsConnector::default();
+                let tls_stream = connector.connect(host, tcp_stream).await.unwrap();
+                break MaybeTlsStream::from(tls_stream);
             }
             _ => (),
         }
     };
 
+    #[cfg(any(feature = "tokio-native-tls", feature = "tokio-rustls"))]
+    let mut maybe_tls_stream = maybe_tls_stream.compat();
```

Comment:
This is where `tokio_util::compat` can be used, which allow users to use the same `futures_io::Async{Read,Write}` accross runtimes.

That's also the reason why proposing a concrete `TcpStream` could be interesting: it would make the compat transparent for them.